shopfloor_mobile_base/static/wms/src/components/searchbar/searchbar.js

```diff
@@ -33,18 +36,76 @@ Vue.component("searchbar", {
             type: Boolean,
             default: true,
         },
+        // remove leading/trailing spaces from input before searching
+        autotrim: {
+            type: Boolean,
+            default: true,
+        },
+        // on scanned input without end of line, the search will run after 50ms. Set to 0 to disable
+        autosearch: {
+            type: Number,
+            default: 50,
+        },
+        // on manually typed input, the search will run after time. Set to 1500 (ms) as time for typing. By default, disabled, enter must be pressed
+        autosearch_typing: {
+            type: Number,
+            default: 0,
+        },
     },
     mounted: function() {
         this.$root.event_hub.$on("screen:reload", this.on_screen_reload);
     },
+    computed: {
+        // defined as computed property to put a new instance in cache each
+        // time the reactive debounceWait is modified
+        debouncedSearch() {
+            return _.debounce(function(e) {
+                if (
+                    this.entered.length == 1 &&
+                    this.debounceWait != this.autosearch_typing
+                ) {
+                    this.debounceWait = this.autosearch_typing;
+                    if (!this.debounceWait) return;
+                    return this.debouncedSearch();
+                }
+                if (!this.debounceWait) return;
+                return this.search(e);
+            }, this.debounceWait);
+        },
+    },
+    watch: {
+        entered: function(val) {
+            if (this.autotrim) {
+                let trimmed = val.trim();
+                if (trimmed !== val) {
+                    this.entered = trimmed;
+                    return;
+                }
+            }
+            if (!this.autosearch) return;
+            if (val.length == 0) {
+                this.debouncedSearch.cancel();
+                this.debounceWait = this.autosearch;
+                return;
+            }
+            return this.debouncedSearch();
+        },
+    },
     methods: {
         search: function(e) {
-            e.preventDefault();
             // Talk to parent
+            if (!this.entered) return;
             this.$emit("found", {
                 text: this.entered,
-                type: e.target.dataset.type,
+                type: String,
```

Comment:
Please restore this line. The type here is the type of event, for internal usage.
When you bind to `found` you can pass a specific type through the data attribute to mark the event w/ a specific "type".
Call it `kind` if you want ;)

Reply:
check where/how this is used https://github.com/OCA/wms/search?q=input_data_type

Reply:
fixed, thanks

Reply:
Hmm, now it is broken. I need to investigate

Reply:
solved